engine/crates/engine-v2/src/sources/graphql/mod.rs

```diff
@@ -5,5 +5,45 @@ mod request;
 mod root_fields;
 mod subscription;
 
+use std::time::Duration;
+
 pub(crate) use federation::*;
+use headers::HeaderMapExt;
 pub(crate) use root_fields::*;
+
+use grafbase_telemetry::gql_response_status::GraphqlResponseStatus;
+use runtime::bytes::OwnedOrSharedBytes;
+
+fn should_update_cache(status: GraphqlResponseStatus, cache_control: Option<&headers::CacheControl>) -> bool {
+    status.is_success()
+        && cache_control
+            .map(|cache_control| !(cache_control.private() || cache_control.no_store()))
+            .unwrap_or(true)
+}
+
+fn calculate_cache_ttl(
```

Comment:
would have created a `graphql/entity_cache.rs` or something like this

Reply:
well I mean, you didn't :D If you intended to do later then great. Just saying what comes to mind reading the PR once marked as reviewable